src/overlays/actors/ovl_Obj_Um/z_obj_um.c

```diff
@@ -46,131 +80,1951 @@ const ActorInit Obj_Um_InitVars = {
+    Player* player = GET_PLAYER(play);
+
+    player->stateFlags1 |= 0x20;
+    ObjUm_SetPlayerPosition(this, play);
+    ObjUm_RotatePlayer(this, play, 0);
+    this->flags |= OBJ_UM_FLAG_0004;
+    ObjUm_UpdateAnim(this, play, 2);
+
+    if (ActorCutscene_GetCanPlayNext(this->dyna.actor.cutscene)) {
+        ActorCutscene_StartAndSetUnkLinkFields(this->dyna.actor.cutscene, &this->dyna.actor);
+        ObjUm_SetupAction(this, ObjUm_PostMilkRunWaitPathFinished);
+    } else {
+        ActorCutscene_SetIntentToPlay(this->dyna.actor.cutscene);
+    }
+}
+
+void ObjUm_TerminaFieldIdle(ObjUm* this, PlayState* play) {
+    ObjUm_UpdateAnim(this, play, 2);
+    SkelAnime_Update(&this->skelAnime);
+}
+
+void func_80B7AF30(ObjUm* this, PlayState* play) {
+    if (!(this->flags & OBJ_UM_FLAG_0001)) {
+        this->dyna.actor.shape.rot.x = 0;
+        this->dyna.actor.shape.rot.z = 0;
+    } else {
+        CollisionPoly* sp44;
+        s32 pad;
+        s32 pad2;
+        Vec3f sp30;
+
+        this->flags &= ~OBJ_UM_FLAG_0001;
+
+        if (1) {}
+
+        this->unk_2D0 = this->unk_2C4;
+        this->unk_2E8 = this->unk_2DC;
+
+        this->unk_2D0.y = BgCheck_EntityRaycastFloor1(&play->colCtx, &sp44, &this->unk_2C4);
+        if (this->unk_2D0.y == BGCHECK_Y_MIN) {
+            this->unk_2D0.y = this->dyna.actor.floorHeight;
+        }
+
+        if (1) {}
+
+        this->unk_2E8.y = BgCheck_EntityRaycastFloor1(&play->colCtx, &sp44, &this->unk_2DC);
+        if (this->unk_2E8.y == BGCHECK_Y_MIN) {
+            this->unk_2E8.y = this->dyna.actor.floorHeight;
+        }
+
+        sp30.x = (this->unk_2D0.x + this->unk_2E8.x) * 0.5f;
+        sp30.y = (this->unk_2D0.y + this->unk_2E8.y) * 0.5f;
+        sp30.z = (this->unk_2D0.z + this->unk_2E8.z) * 0.5f;
+
+        this->dyna.actor.shape.rot.x =
+            (s16)Math_Atan2S(sp30.y - this->dyna.actor.world.pos.y, sqrtf(SQ(sp30.x - this->dyna.actor.world.pos.x) +
+                                                                          SQ(sp30.z - this->dyna.actor.world.pos.z)));
+        this->dyna.actor.shape.rot.z = (s16)-Math_Atan2S(
+            sp30.y - this->unk_2D0.y, sqrtf(SQ(sp30.x - this->unk_2D0.x) + SQ(sp30.z - this->unk_2D0.z)));
+        if (this->flags & OBJ_UM_FLAG_MOVING) {
+            this->dyna.actor.shape.rot.x += BINANG_SUB((Rand_ZeroOne() * 100.0f), 50.0f);
+            this->dyna.actor.shape.rot.z += BINANG_SUB((Rand_ZeroOne() * 100.0f), 50.0f);
+        }
+    }
+}
+
+void ObjUm_StopAnim(ObjUm* this, PlayState* play) {
+    Animation_PlayOnce(&this->skelAnime, &object_um_Anim_012CC0);
+    this->unk_304 = 0;
+}
+
+typedef struct {
+    /* 0x00 */ AnimationHeader* anim;
+    /* 0x04 */ s32 animMoves;
+} struct_80B7C25C; // size = 0x08
+
+struct_80B7C25C D_80B7C25C[] = {
+    { &object_um_Anim_012CC0, true },  { &object_um_Anim_01213C, true },
+    { &object_um_Anim_019E10, false }, { NULL, false },
+    { &object_um_Anim_0126C4, false },
+};
```

Comment:
```suggestion
struct_80B7C25C D_80B7C25C[] = {
    { &object_um_Anim_012CC0, true },  { &object_um_Anim_01213C, true },
    { &object_um_Anim_019E10, false },
    { &object_um_Anim_0126C4, false },
};
```
Are you sure this isn't just padding? I think even adding a:
`s32 D_80B7C27C = 0` might be better?

Reply:
It can't be padding since it is not at the end of the array. I had that same issue when reading this array today.
I formatted it a bit to make this obvious:
```c

struct_80B7C25C sUmAnims[] = {
    { &gUmTrotAnim, true },      // OBJ_UM_ANIM_TROT
    { &gUmGallopAnim, true },    // OBJ_UM_ANIM_GALLOP
    { &gUmIdleAnim, false },     // OBJ_UM_ANIM_IDLE
    { NULL, false },             // OBJ_UM_ANIM_3
    { &gUmLookBackAnim, false }, // OBJ_UM_ANIM_LOOK_BACK
};

```

Reply:
Oh funny, yeah I totally read that wrong